serpentTools/parsers/fissionMatrix.py

```diff
@@ -1,15 +1,326 @@
+
+        Parameters
+        ----------
+        title: str
+            Plot title
+        xlabel: str
+            x-axis label
+        ylabel: str
+            y-axis label
+        cmap: str
+            Color map
+        logScale: bool
+            Log-scale on, True; Log-scale off, False
+        cbarLabel: None or str
+            Label to apply to colorbar
+
+        """
+        self._matPlot(self.fMat, ax, title, xlabel, ylabel, cmap, logScale,
+                      cbarLabel)
+
+    def fMatUPlot(self, ax=None, title=None, xlabel=None, ylabel=None, cmap=None,
+                  logScale=False, cbarLabel=None):
+        """
+        Plots sparsity pattern of the uncertainty matrix associated to
+        fission matrix entries
+
+        Parameters
+        ----------
+        title: str
+            Plot title
+        xlabel: str
+            x-axis label
+        ylabel: str
+            y-axis label
+        cmap: str
+            Color map
+        logScale: bool
+            Log-scale on, True; Log-scale off, False
+        cbarLabel: None or str
+            Label to apply to colorbar
+
+        """
+        self._matPlot(self.fMatU, ax, title, xlabel, ylabel, cmap, logScale,
+                      cbarLabel)
+
+    def _matPlot(self, A, ax=None, title=None, xlabel=None, ylabel=None, cmap=None,
+                 logScale=False, cbarLabel=None):
+        v = A.shape
+        xticks = range(0, v[0])
+        ax = cartMeshPlot(A, xticks, xticks[::-1], ax=ax, logScale=logScale,
+                          cbarLabel=cbarLabel)
+        formatPlot(ax, title=title, xlabel=xlabel, ylabel=ylabel, cmap=cmap)
+        plt.xticks([])
+        plt.yticks([])
+
+    def fMatEig(self):
+        """
+        Computes the Fission Matrix Eigenpairs
+
+        Returns
+        -------
+            self.domEigVal: float
+                Dominant Eigenvalue
+            self.domEigVec: np.array
+                Dominant Eigenvector
+
+        """
+        eigValVec, eigVecMat = np.linalg.eig(self.fMat)
+        self.eigVecMat = self.eigVecNorm(eigVecMat)
+        self.domEigVal = eigValVec[0]
+        self.eigValVec = eigValVec
+        self.domEigVec = self.eigVecMat[:, 0]
+        self.domRatio = self.eigValVec[1] / self.eigValVec[0]
+        return self.domEigVal, self.domEigVec
+
+    def eigVecNorm(self, w):
+        # Flips and normalizes eigenvectors
+        z = w[:, 0]
+        lunghezza = len(z)
+        cont = len(z[z < 0])
+        if cont == lunghezza or cont == 0:
+            w = w / np.sum(w[:, 0])
+        else:
+            error('The Dominant Eigenvector is not Positive')
+        return w
+
+    def eigVecPlot(self, eigNum=1, xdata=None, ax=None,
+                   title='Neutron fission source', xlabel=None, ylabel=None,
+                   grid=True):
+        """
+        Plots the spatial distribution of the eigNum-th mode
+
+        Parameters
+        ----------
+        eigNum: list
```

Comment:
Can you make this method work for plotting one mode by passing an integer, or plotting all modes by not passing anything?

All modes would be very busy for large arrays, but would be easier than passing a list of all indices in the array.